cpp/ycm/tests/CMakeLists.txt

```diff
@@ -77,13 +82,39 @@ add_executable( ${PROJECT_NAME}
                 ${SOURCES}
               )
 
+if ( MSVC )
+  # This is needed to compile tests with the gtest shared library
+  set_target_properties( ${PROJECT_NAME}
+                         PROPERTIES
+                         COMPILE_DEFINITIONS
+                         "GTEST_LINKED_AS_SHARED_LIBRARY=1" )
+
+  # Preprocessor definition used to export symbols for gmock tests
+  set_target_properties( ${SERVER_LIB} PROPERTIES DEFINE_SYMBOL "TESTS_EXPORTS" )
+
+  # Fix gtest build on MSVC 11. See:
+  # http://stackoverflow.com/questions/8274588/c2977-stdtuple-too-many-template-arguments-msvc11
```

Comment:
SO has short links for all answers and questions through "share" links (bottom of each question/answer). Maybe use that?